internal/core/src/exec/expression/Expr.h

```diff
@@ -1218,9 +1221,16 @@ class SegmentExpr : public Expr {
         }
     }
 
+    bool
+    CanUseIndex() const {
+        // Ngram index should be used in specific execution path (CanUseNgramIndex -> ExecNgramMatch).
+        // TODO: if multiple indexes are supported, this logic should be changed
+        return is_index_mode_ && !CanUseNgramIndex();
```

Comment:
we can remove member `is_index_mode_`, and replace with `num_index_chunk_ != 0`?

Reply:
done